elements/pfe-primary-detail/src/pfe-primary-detail.js

```diff
@@ -0,0 +1,177 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+
+class PfePrimaryDetail extends PFElement {
+  static get tag() {
+    return "pfe-primary-detail";
+  }
+
+  static get meta() {
+    return {
+      title: "Primary detail",
+      description: ""
+    };
+  }
+
+  get templateUrl() {
+    return "pfe-primary-detail.html";
+  }
+
+  get styleUrl() {
+    return "pfe-primary-detail.scss";
+  }
+
+  // static get events() {
+  //   return {
+  //   };
+  // }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Container;
+  }
+
+  static get properties() {
+    return {};
+  }
+
+  constructor() {
+    super(PfePrimaryDetail, { type: PfePrimaryDetail.PfeType });
+
+    this._handleHideShow = this._handleHideShow.bind(this);
+
+    this._detailsNav = this.shadowRoot.getElementById("details-nav");
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // If you need to initialize any attributes, do that here
+
+    this._slots = {
+      detailsNav: this.shadowRoot.querySelector('slot[name="details-nav"]').assignedElements(),
+      details: this.shadowRoot.querySelector('slot[name="details"]').assignedElements(),
+      detailsNavHeader: this.shadowRoot.querySelector('slot[name="details-nav--header"]').assignedElements(),
+      detailsNavFooter: this.shadowRoot.querySelector('slot[name="details-nav--footer"]').assignedElements()
+    };
+
+    this._detailsWrapper = this.shadowRoot.getElementById("details-wrapper");
+
+    // Add appropriate markup and behaviors
+    this._scanLightDom();
+    // Set first item as active for initial load
+    this._handleHideShow({ target: this._slots.detailsNav[0] });
+  }
+
+  // @todo Add mutation observer in case any content changes, rerun scanLightDom?
+
+  disconnectedCallback() {
+    for (let index = 0; index < this._slots.detailsNav.length; index++) {
+      this._slots.detailsNav[index].removeEventListener("click", this._handleHideShow);
+    }
+  }
+
+  /**
+   * Create nav functionality and adds additional HTML/attributes to markup
+   */
+  _scanLightDom() {
+    if (this._slots.detailsNav.length !== this._slots.details.length) {
+      console.error(
+        `${this.tag}: The number of item headings does not match the number of item details. Found ${this._slots.detailsNav.length} item headings & ${this._slots.details.length} item details.`
+      );
+      return;
+    }
+
+    // Setup left sidebar navigation
+    this._slots.detailsNav.forEach((toggle, index) => {
+      let attr = toggle.attributes;
+      toggle.dataset.index = index;
+
+      const button = document.createElement("button");
+
+      button.innerText = toggle.textContent;
+      // Copy over attributes from original element
+      [...attr].forEach(toggle => {
```

Comment:
Potentially want to run these through an `allowlist` or `denylist` to prevent `style` tags or anything else problematic from coming over.  cc @kylebuch8 re: pfe-button does this.